src/parser/jobs/pld/modules/Atonement.tsx

```diff
@@ -0,0 +1,80 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {Analyser} from 'parser/core/Analyser'
+import Checklist, {TieredRule, TARGET, Requirement} from 'parser/core/modules/Checklist'
+import {Data} from 'parser/core/modules/Data'
+import React from 'react'
+import {dependency} from '../../../core/Injectable'
+
+const STACKS_GAINED = 3
+
+// In seconds
+const UPTIME = {
+	95: TARGET.WARN,
+	100: TARGET.SUCCESS,
+}
+
+export default class Atonement extends Analyser {
+	static override handle = 'Atonement'
+	static override title = t('pld.Atonement.title')`Atonement`
+
+	@dependency private checklist!: Checklist
+	@dependency private data!: Data
+
+	protected actions: number = 0
+	protected buffs: number = 0
+
+	override initialise() {
+
+		this.addEventHook({
+			type: 'action',
+			source: this.parser.actor.id,
+			action: this.data.actions.ATONEMENT.id,
+		}, this.onAtonement)
```

Comment:
weird personal thought, wouldn't this be better?

```ts

		this.addEventHook({
			type: 'action',
			source: this.parser.actor.id,
			action: this.data.actions.ATONEMENT.id,
		}, () => this.actions++)
```

it'd save you a whole function, anyway?

Same with the `onRoyal` part